app/assets/stylesheets/base/typography.scss

```diff
@@ -96,27 +105,79 @@ body {
   h5,
   h6 {
     font-family: var(--ff-sans-serif);
-    margin: var(--content-rythm) 0;
-    margin-top: calc(var(--content-rythm) * 2);
+    margin: calc(var(--content-rhythm) * 2) 0 calc(var(--content-rhythm) / 2);
     padding: 0;
+
+    & + p,
+    & + ul {
+      margin-top: 0;
+    }
```

Comment:
We should probably find more bullet-proof way of doing this since, our posts can have literally almost any HTML elements, including divs, iframes, pres, codes, images, etc...

Reply:
Yeah I agree with you, I was attempting to write `& + *`, but that didn't seem to work — hah!

Do you have an alternative solution for this area?